packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/toolbox.tsx

```diff
@@ -0,0 +1,165 @@
+    toolbox_dom: HTMLElement;
+};
+
+const Toolbox = ({
+    onMount,
+    onUnmount,
+    hasSubCategory,
+    is_mobile,
+    is_search_loading,
+    onSearch,
+    onSearchBlur,
+    onSearchClear,
+    onSearchKeyUp,
+    onToolboxItemClick,
+    onToolboxItemExpand,
+    sub_category_index,
+    toolbox_dom,
+}: TToolbox) => {
+    const toolbox_ref = React.useRef(ToolboxItems);
+    const [is_open, setOpen] = React.useState(true);
+
+    React.useEffect(() => {
+        onMount(toolbox_ref);
+        return () => onUnmount();
+    }, [onMount, onUnmount]);
+
+    if (is_mobile) {
+        return null;
+    }
+
+    return (
+        <div className='dashboard__toolbox'>
+            <div id='gtm-toolbox' className='db-toolbox__content'>
+                <div className='db-toolbox__header'>
+                    <div className='db-toolbox__title' onClick={() => setOpen(!is_open)}>
+                        {localize('Blocks menu')}
+                        <span
+                            className={classNames('db-toolbox__title-chevron', {
+                                'db-toolbox__title-chevron--active': is_open,
+                            })}
+                        >
+                            <Icon icon='IcChevronDownBold' />
+                        </span>
+                    </div>
+                </div>
+                <div
+                    className={classNames('db-toolbox__content-wrapper', {
+                        'db-toolbox__content-wrapper--active': is_open,
+                    })}
+                >
+                    <SearchBox
+                        is_search_loading={is_search_loading}
+                        onSearch={onSearch}
+                        onSearchBlur={onSearchBlur}
+                        onSearchClear={onSearchClear}
+                        onSearchKeyUp={onSearchKeyUp}
+                    />
+                    <div className='db-toolbox__category-menu'>
+                        {toolbox_dom &&
+                            (Array.from(toolbox_dom.childNodes) as HTMLElement[]).map((category, index) => {
+                                if (category.tagName.toUpperCase() === 'CATEGORY') {
+                                    const has_sub_category = hasSubCategory(category.children);
+                                    const is_sub_category_open = sub_category_index.includes(index);
+                                    return (
+                                        <div
+                                            key={`db-toolbox__row--${category.getAttribute('id')}`}
+                                            className='db-toolbox__row'
+                                        >
+                                            <div
+                                                className='db-toolbox__item'
+                                                onClick={() => {
+                                                    // eslint-disable-next-line no-unused-expressions
```

Comment:
We don't need it any more... so I removed it. ✅

Reply:
![Screenshot 2022-09-14 at 4 55 24 PM](https://user-images.githubusercontent.com/90243468/190109259-c96038af-9913-4ba5-8d43-bb4e5bf978e7.png)


this was there to prevent from this error... and I had to add it back